### DISCUSSION TITLE
매니지드, 언매니지드 언어란 무엇인가요?

### DISCUSSION THREAD

**Discussion by pipisebastian** [publdaze/js-study#28] [변수]
- 매니지드, 언매니지드 언어란 무엇인가요?
  - 각각의 장단점을 설명해주세요.

**Comment by pipisebastian** [comment]
> 매니지드, 언매니지드 언어란 무엇인가요?
각각의 장단점을 설명해주세요.

프로그래밍 언어는 메모리를 관리하는 방식에 따라 2가지로 나뉘어지는데요, 

언매니지드 언어는 개발자가 명시적으로 메모리를 제어할 수 있습니다. 매니지드 언어는 c언어는, malloc, free등으로 메모리를 할당, 수거할 수 있습니다. 
⇒ 성능 향상 / 오류 발생 가능

매니지드 언어는 언어 자체가 메모리를 관리합니다. 예를 들어 자바스크립트의 가바지 콜렉터는 더이상 사용하지 않는 메모리를 알아서 해제해줍니다. 
⇒ 성능 감소 / 개발자 관리 영역 감소.

**Comment by publdaze** [comment]
> - 매니지드, 언매니지드 언어란 무엇인가요?
- 매니지드 언어는 메모리에 직접 접근이 가능. C언어는 malloc으로 메모리 할당하고 free로 메모리 해제할 수 있음
- 언매니지드 언어는 메모리에 직접 접근 불가.
> - 각각의 장단점을 설명해주세요.
- 매니지드 언어는 메모리를 직접 접근할 수 있어서 메모리를 효율적으로 관리 가능하지만 메모리에 직접 접근해서 위험 부담이 있음
- 언매니지드 언어는 직접 접근할 수 없어서 엔진에 의존적으로 메모리 관리를 맡겨야하지만, 안정성은 보장됨
